### PR TITLE
Feat: Char validator

### PR DESCRIPTION
**Pull request by Meldiron**

### CODE REVIEW

File: src/Validator/Char.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+use Utopia\Validator;
+
+/**
+ * Text
+ *
+ * Validate that an variable is a valid text value
+ *
+ * @package Utopia\Validator
+ */
+class Char extends Validator
```
Comment (eldadfux) [inline src/Validator/Char.php:24]:
```suggestion
class Chars extends Validator
```

Make sense?

**Comment by kodumbeats** [comment]
To my 👀, this looks to be the Text validator plus a whitelist of chars. Any reason we couldn't just add this functionality there?

`new Text(255, [
    ...Text::ALPHABET_LOWER,
    ...Text::ALPHABET_UPPER,
    ...Text::NUMBERS
]`

Where the constructor has a default, empty value for a whitelist
`public function __construct(int $length, array $whitelist = []){}`

**Comment by Meldiron** [comment]
@eldadfux @kodumbeats I decided to go with Brandon solution of merging this into a Text validator. Whitelist is optional and ignored if it's an empty array (who would want no allowed characters anyway).

Please revisit the PR 😇

**Review by eldadfux — APPROVED** [review]

File: src/Validator/Text.php [hunk 2]
```diff
@@ -23,21 +23,33 @@
...
+
     /**
      * @var int
      */
     protected int $length = 0;
 
+    /**
+     * @var string[]
+     */
+    protected array $whitelist = [];
```
Comment (christyjacob4) [inline src/Validator/Text.php:38]: we want to gradually move away from the convention of using `whitelist` in favour of `allowList`

Maybe this PR can be a start to that. WDYT ?

**Pull request merged**